src/App.tsx

```diff
@@ -15,35 +16,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortOption {
+  default = 'default',
+  alphabetically = 'alphabetically',
+  byLength = 'byLength',
+}
+
+interface SortParams {
+  sortOption: SortOption;
+  isReversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortOption, isReversed }: SortParams,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortOption) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortOption) {
+        case SortOption.alphabetically:
+          return good1.localeCompare(good2);
+        case SortOption.byLength:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortOption, setSortOption] = useState<SortOption>(SortOption.default);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const handleReset = () => {
+    setSortOption(SortOption.default);
+    setIsReversed(false);
+  };
```

Comment:
The function's name 'handleReset' does not start with a verb that clearly indicates its purpose. Consider renaming it to something like 'resetSortAndOrder' to better reflect what the function does, adhering to the checklist item 4: Function's name should start with a verb.